tvb_epilepsy/service/probability_distribution_factory.py

```diff
@@ -0,0 +1,168 @@
+
+from collections import OrderedDict
+
+import numpy as np
+from scipy.optimize import minimize
+
+from tvb_epilepsy.base.utils.log_error_utils import raise_value_error, warning
+from tvb_epilepsy.base.utils.data_structures_utils import isequal_string, dicts_of_lists_to_lists_of_dicts
+
+
+AVAILABLE_DISTRIBUTIONS = ["uniform", "normal", "gamma", "lognormal", "exponential", "beta", "chisquare",
+                           "binomial", "poisson", "bernoulli"]
+
+CONSTRAINT_ABS_TOL = 0.001
+
+
+def generate_distribution(distrib_type, target_shape=None, loc=0.0, scale=1.0, use="manual", **target):
+    if np.in1d(distrib_type.lower(), AVAILABLE_DISTRIBUTIONS):
+        exec("from tvb_epilepsy.base.model.statistical_models.probability_distributions."
+             + distrib_type.lower() + "_distribution import " + distrib_type.title() + "Distribution")
+        distribution = eval(distrib_type.title() + "Distribution()")
```

Comment:
This should turn into a factory method that uses actual import/constructor calls.